op-batcher/batcher/channel_manager.go

```diff
@@ -301,15 +304,21 @@ func (s *channelManager) registerL1Block(l1Head eth.BlockID) {
 func (s *channelManager) processBlocks() error {
 	var (
 		blocksAdded int
-		_chFullErr  *ChannelFullError // throw away, just for type checking
 		latestL2ref eth.L2BlockRef
 	)
-	for i, block := range s.blocks {
+	for i := 0; ; i++ {
+		block, ok := s.blocks.Dequeue()
+		if !ok {
+			break
+		}
+
 		l1info, err := s.currentChannel.AddBlock(block)
-		if errors.As(err, &_chFullErr) {
+		if errors.Is(err, new(ChannelFullError)) {
```

Comment:
Does this actually work? I thought `errors.Is` can only be used with fixed error variables, so those that by convention start with `Err...`, whereas `errors.As` has to be used with error _types_, and then we need to pass a double-pointer.

Reply:
Good catch, this actually does not work. I changed this back.